docker/tarfile/dest.go

```diff
@@ -256,20 +247,18 @@ func (d *Destination) PutManifest(ctx context.Context, m []byte, instanceDigest
 		repoTags = append(repoTags, refString)
 	}
 
-	items := []ManifestItem{{
+	d.manifest = append(d.manifest, ManifestItem{
 		Config:       man.ConfigDescriptor.Digest.Hex() + ".json",
 		RepoTags:     repoTags,
 		Layers:       layerPaths,
 		Parent:       "",
 		LayerSources: nil,
-	}}
-	itemsBytes, err := json.Marshal(&items)
-	if err != nil {
-		return err
-	}
+	})
 
-	// FIXME? Do we also need to support the legacy format?
-	return d.sendBytes(manifestFileName, itemsBytes)
+	// Reset the repoTags to prevent them from leaking into a following
+	// image/manifest.
+	d.repoTags = []reference.NamedTagged{}
+	return nil
```

Comment:
… it’s also not documented that the caller is supposed to use (only) `AddRepoTags` after each `PutManifest` AFAICS.